### PR TITLE
Added optional filter query param for instance listing

### PR DESCRIPTION
**Pull request by umbreak**

### CODE REVIEW

**Comment by umbreak** [comment]
retest this please

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala [hunk 1]
```diff
@@ -50,40 +50,48 @@ class InstanceRoutes(
...
           val domainId = DomainId(orgId, domain)
           pathEndOrSingleSlash {
-            (get & parameter('from.as[Int] ? pagination.from) & parameter('size.as[Int] ? pagination.size) & parameter('deprecated.as[Boolean].?)) { (from, size, deprecated) =>
-              traceName("listInstancesOfDomain") {
-                val filter = Filter(deprecatedOrNoop(deprecated))
-                buildResponse(instanceQueries.list(domainId, filter, Pagination(from, size)), Pagination(from, size))
+            (get & paginatedAndFiltered) { (pagination, filterOpt) =>
+              parameter('deprecated.as[Boolean].?) { deprecated =>
+                traceName("listInstancesOfDomain") {
+                  val filter = Filter(deprecatedAndRev(deprecated)) and filterOpt.map(_.expr) and domExpr(domainId)
```
Comment (bogdanromanx) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala]: remove `domExpr(domainId)`

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala [hunk 2]
```diff
@@ -50,40 +50,48 @@ class InstanceRoutes(
...
+                  buildResponse(instanceQueries.list(orgId, filter, pagination), pagination)
```
Comment (bogdanromanx) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala]: orgId => domainId

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala [hunk 3]
```diff
@@ -50,40 +50,48 @@ class InstanceRoutes(
...
           pathPrefix(Segment) { name =>
             pathEndOrSingleSlash {
-              (get & parameter('from.as[Int] ? pagination.from) & parameter('size.as[Int] ? pagination.size) & parameter('deprecated.as[Boolean].?)) { (from, size, deprecated) =>
-                traceName("listInstancesOfSchemaName") {
-                  val filter = Filter(deprecatedOrNoop(deprecated))
-                  buildResponse(instanceQueries.list(domainId, name, filter, Pagination(from, size)), Pagination(from, size))
+              (get & paginatedAndFiltered) { (pagination, filterOpt) =>
+                parameter('deprecated.as[Boolean].?) { deprecated =>
+                  traceName("listInstancesOfSchemaName") {
+                    val filter = Filter(deprecatedAndRev(deprecated)) and filterOpt.map(_.expr) and domExpr(domainId) and schemaNameExpr(name)
```
Comment (bogdanromanx) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala]: ditto

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala [hunk 4]
```diff
@@ -95,10 +103,12 @@ class InstanceRoutes(
...
 
                   pathEndOrSingleSlash {
-                    (get & parameter('from.as[Int] ? pagination.from) & parameter('size.as[Int] ? pagination.size) & parameter('deprecated.as[Boolean].?)) { (from, size, deprecated) =>
-                      traceName("listInstancesOfSchema") {
-                        val filter = Filter(deprecatedOrNoop(deprecated))
-                        buildResponse(instanceQueries.list(schemaId, filter, Pagination(from, size)), Pagination(from, size))
+                    (get & paginatedAndFiltered) { (pagination, filterOpt) =>
+                      parameter('deprecated.as[Boolean].?) { deprecated =>
+                        traceName("listInstancesOfSchema") {
+                          val filter = Filter(deprecatedAndRev(deprecated)) and filterOpt.map(_.expr) and domExpr(domainId) and schemaExpr(schemaId)
```
Comment (bogdanromanx) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/InstanceRoutes.scala]: ditto

**Comment by umbreak** [comment]
retest this please

**Pull request merged**